File: languages/java/src/main/java/com/bloock/sdk/entity/identity/CredentialReceipt.java
package com.bloock.sdk.entity.identity;

import com.bloock.sdk.bridge.proto.IdentityEntities;

/**
 * Represents a receipt for a credential, including the credential itself, its ID, and type.
 */
public class CredentialReceipt {
  private final Credential credential;
  private final String credentialId;
  private final String credentialType;

  /**
   * Constructs an CredentialReceipt object with the specified parameters.
   * @param credential
   * @param credentialId
   * @param credentialType
   */
  public CredentialReceipt(Credential credential, String credentialId, String credentialType) {
    this.credential = credential;
    this.credentialId = credentialId;
    this.credentialType = credentialType;
  }

  public static CredentialReceipt fromProto(IdentityEntities.CredentialReceipt res) {
    return new CredentialReceipt(
        Credential.fromProto(res.getCredential()), res.getCredentialId(), res.getCredentialType());
  }

  public IdentityEntities.CredentialReceipt toProto() {
    return IdentityEntities.CredentialReceipt.newBuilder()
        .setCredential(this.credential.toProto())
        .setCredentialId(this.credentialId)
        .setCredentialType(this.credentialType)
        .build();
  }

  /**
   * Gets the credential object.
   * @return
   */
  public Credential getCredential() {
    return credential;
  }

  /**
   * Gets the ID associated with the credential.
   * @return
   */
  public String getCredentialId() {
    return credentialId;
  }

  /**
   * Gets the type of the credential.
   * @return
   */
  public String getCredentialType() {
    return credentialType;
  }
}

File: languages/go/entity/identityV2/credential_receipt.go
package identityV2

import "github.com/bloock/bloock-sdk-go/v2/internal/bridge/proto"

// CredentialReceipt represents a receipt for a credential, including the credential itself, its ID, and type.
type CredentialReceipt struct {
	Credential     Credential
	CredentialId   string
	CredentialType string
}

func NewCredentialReceiptFromProto(s *proto.CredentialReceiptV2) CredentialReceipt {
	if s == nil {
		return CredentialReceipt{}
	}

	return CredentialReceipt{CredentialId: s.CredentialId, CredentialType: s.CredentialType, Credential: NewCredentialFromProto(s.Credential)}
}

func (c CredentialReceipt) ToProto() *proto.CredentialReceiptV2 {
	return &proto.CredentialReceiptV2{
		Credential:     c.Credential.ToProto(),
		CredentialId:   c.CredentialId,
		CredentialType: c.CredentialType,
	}
}
